Support git blame ignore revs

terrdavis opened the pull request:
In response to this [comment](https://github.com/ipython/ipython/issues/11816#issuecomment-554671200).

Please note:
- neither Github nor GitLab supports this feature.
- the IntelliJ plugin [GitToolBox](https://github.com/zielu/GitToolBox) supports this feature, so perhaps other client-side git tools do too?

terrdavis committed:
Adds support for blame ignore.

Windows and non-Windows scripts are provided for initial setup.
Commit hashes to ignore are entered in the file .git-blame-ignore-revs

terrdavis committed:
Add section to README about enabling blame ignore revs.

Carreau commented:
Thanks ! That is great. 

Even if it's not (yet) supported by github/gitlab, when we do such commits we can link to this documentation explicitly, and see if some clarifications are needed. 

I'll need to fix MANIFEST.in to `exclude .git-blame-ignore-revs` or test are failling; i'll try to do that later unless you beat me to it.

Carreau committed:
Exclude .git-blame-ignore-revs to fix CI

.git-blame-ignore-revs is added so that  we can ignore revisions in .git-blame,  mostly code reformatting; but CI check that all files are explicitly included (or excluded) from manifest; so we need to list this one explicitly for travis to succeed.

Carreau commented:
Thanks !
